Something still isn't working right. I checked this out in a fully-built openpilot, and had to run twice before everything was built properly:
```
batman:lib$ scons -u -j8 .
scons: Entering directory `/home/batman/twopilot'
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
Retrieved `selfdrive/controls/lib/longitudinal_mpc_lib/c_generated_code/Makefile' from cache
Retrieved `selfdrive/controls/lib/longitudinal_mpc_lib/c_generated_code/main_long.c' from cache
Retrieved `selfdrive/controls/lib/longitudinal_mpc_lib/c_generated_code/acados_solver_long.h' from cache
Retrieved `selfdrive/controls/lib/longitudinal_mpc_lib/c_generated_code/acados_solver.pxd' from cache
Retrieved `selfdrive/controls/lib/longitudinal_mpc_lib/c_generated_code/long_model/long_expl_vde_adj.c' from cache
Retrieved `selfdrive/controls/lib/longitudinal_mpc_lib/c_generated_code/long_model/long_model.h' from cache
cd /home/batman/twopilot/selfdrive/controls/lib/longitudinal_mpc_lib && python3 long_mpc.py
Warning: Please note that the following versions of CasADi  are officially supported: 3.5.6 or 3.5.5 or 3.5.4 or 3.5.3 or 3.5.2 or 3.5.1 or 3.4.5 or 3.4.0.
 If there is an incompatibility with the CasADi generated code, please consider changing your CasADi version.
Version 3.6.3 currently in use.
scons: done building targets.
batman:lib$ scons -u -j8 .
scons: Entering directory `/home/batman/twopilot'
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
clang -o selfdrive/controls/lib/longitudinal_mpc_lib/c_generated_code/acados_ocp_solver_pyx.o -c -std=gnu11 -DSWAGLOG="\"common/swaglog.h\"" -g -fPIC -O2 -Wunused -Wshadow -Wno-unknown-warning-option -Wno-deprecated-register -Wno-register -Wno-inconsistent-missing-override -Wno-c99-designator -Wno-reorder-init-list -Wno-error=unused-but-set-variable -DSWAGLOG="\"common/swaglog.h\"" -Wno-#warnings -Wno-shadow -Wno-deprecated-declarations -I. -Ithird_party/acados/include -Ithird_party/acados/include/blasfeo/include -Ithird_party/acados/include/hpipm/include -Ithird_party/catch2/include -Ithird_party/libyuv/include -Ithird_party/json11 -Ithird_party/linux/include -Ithird_party/snpe/include -Ithird_party/mapbox-gl-native-qt/include -Ithird_party/qrcode -Ithird_party -Icereal -Iopendbc/can -I/home/batman/.pyenv/versions/3.11.4/include/python3.11 -I/home/batman/.pyenv/versions/3.11.4/lib/python3.11/site-packages/numpy/core/include selfdrive/controls/lib/longitudinal_mpc_lib/c_generated_code/acados_ocp_solver_pyx.c
selfdrive/controls/lib/longitudinal_mpc_lib/c_generated_code/acados_ocp_solver_pyx.c:8036:28: warning: equality comparison with extraneous parentheses [-Wparentheses-equality]
  if (((__pyx_v_c_mode[0]) == 'f')) {
       ~~~~~~~~~~~~~~~~~~~~^~~~~~
selfdrive/controls/lib/longitudinal_mpc_lib/c_generated_code/acados_ocp_solver_pyx.c:8036:28: note: remove extraneous parentheses around the comparison to silence this warning
  if (((__pyx_v_c_mode[0]) == 'f')) {
      ~                    ^     ~
selfdrive/controls/lib/longitudinal_mpc_lib/c_generated_code/acados_ocp_solver_pyx.c:8036:28: note: use '=' to turn this equality comparison into an assignment
  if (((__pyx_v_c_mode[0]) == 'f')) {
                           ^~
                           =
1 warning generated.
scons: done building targets.
batman:lib$ scons -u -j8 .
scons: Entering directory `/home/batman/twopilot'
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
scons: `selfdrive/controls/lib' is up to date.
scons: done building targets.
```

--- a/selfdrive/controls/lib/lateral_mpc_lib/SConscript
+++ b/selfdrive/controls/lib/lateral_mpc_lib/SConscript
@@ -38,8 +38,8 @@ generated_files = [
   f'{gen}/lat_model/lat_expl_vde_adj.c',
 
   f'{gen}/lat_model/lat_model.h',
-  f'{gen}/lat_cost/lat_cost.h',
   f'{gen}/lat_constraints/lat_constraints.h',
+  f'{gen}/lat_cost/lat_cost.h',
 ] + build_files
 
 acados_dir = '#third_party/acados'

--- a/selfdrive/controls/lib/lateral_mpc_lib/SConscript
+++ b/selfdrive/controls/lib/lateral_mpc_lib/SConscript
@@ -38,6 +38,7 @@ generated_files = [
   f'{gen}/lat_model/lat_expl_vde_adj.c',
 
   f'{gen}/lat_model/lat_model.h',
+  f'{gen}/lat_cost/lat_cost.h',
   f'{gen}/lat_constraints/lat_constraints.h',
 ] + build_files
 

--- a/selfdrive/controls/lib/longitudinal_mpc_lib/SConscript
+++ b/selfdrive/controls/lib/longitudinal_mpc_lib/SConscript
@@ -46,7 +46,6 @@ generated_files = [
   f'{gen}/long_model/long_model.h',
   f'{gen}/long_constraints/long_constraints.h',
   f'{gen}/long_cost/long_cost.h',
-
 ] + build_files
 
 acados_dir = '#third_party/acados'

